include/nbl/builtin/glsl/ext/FFT/fft.glsl

```diff
@@ -81,71 +79,86 @@ vec2 nbl_glsl_ext_FFT_getPaddedData(in uvec3 coordinate, in uint channel);
 #error "You need to define `nbl_glsl_ext_FFT_getPaddedData` and mark `_NBL_GLSL_EXT_FFT_GET_PADDED_DATA_DEFINED_`!"
 #endif
 
-// Count Leading Zeroes (naive?)
-uint nbl_glsl_ext_FFT_clz(in uint x) 
-{
-    return 31u - findMSB(x);
-}
-
-uint nbl_glsl_ext_FFT_reverseBits(in uint x)
-{
-    return bitfieldReverse(x);
-}
-
 uint nbl_glsl_ext_FFT_calculateTwiddlePower(in uint threadId, in uint iteration, in uint logTwoN) 
 {
     const uint shiftSuffix = logTwoN - 1u - iteration; // can we assert that iteration<logTwoN always?? yes
     const uint suffixMask = (2u << iteration) - 1u;
     return (threadId & suffixMask) << shiftSuffix;
 }
 
-nbl_glsl_complex nbl_glsl_ext_FFT_twiddle(in uint threadId, in uint iteration, in uint logTwoN, in uint N) 
+nbl_glsl_complex nbl_glsl_ext_FFT_twiddle(in uint threadId, in uint iteration, in uint logTwoN) 
 {
     uint k = nbl_glsl_ext_FFT_calculateTwiddlePower(threadId, iteration, logTwoN);
-    return nbl_glsl_expImaginary(-1 * 2 * nbl_glsl_PI * k / N);
+    return nbl_glsl_expImaginary(-1.0f * 2.0f * nbl_glsl_PI * float(k) / (1 << logTwoN));
```

Comment:
you still have an implicit int to float conversion here, because the left hand side to your `/` operator is uint

Reply:
Do you mean the right-hand side?

Reply:
yeah sorry, got distracted.